server/src/test/java/org/apache/druid/curator/BoundedExponentialBackoffRetryWithQuitTest.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.curator;
+
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.test.TestingServer;
+import org.apache.druid.java.util.common.lifecycle.Lifecycle;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.easymock.EasyMock;
+import org.junit.Assert;
+import org.junit.Test;
+
+public final class BoundedExponentialBackoffRetryWithQuitTest
+{
+
+  private static final Logger log = new Logger(BoundedExponentialBackoffRetryWithQuitTest.class);
+
+  /*
```

Comment:
(Optional) no reason why it shouldn't be a Javadoc comment.

Reply:
> could you please link this discussion from the comment? There is no superstition that code should be "self-contained" in Druid, there are many links from the code to Github. It would be easier for readers to navigate. Currently, readers would be forced to dig into Git history to follow this path.

@leventov I don't understand what you mean.

Reply:
I mean that comments in code should say `see see https://github.com/apache/incubator-druid/pull/6740#discussion_r243368737` or, more generally, `see discussions in https://github.com/apache/incubator-druid/pull/6740` somewhere. In other words, the code should back reference to this PR.

Reply:
@leventov Ok, done.

Reply:
@michael-trelinski this is a wrong place. This link should be a part of the comment in `CuratorModule`. Specifically, it should extend your own comment
```java
// It's unknown whether or not this precaution is needed.  Tests revealed that this path was never taken.
```

Reply:
Apologies.  It's part of the thread-view for BoundedExponentialBackoffRetryWithQuitTest.java.